### PR TITLE
Update master php testing versions

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

**Review by mejo- — APPROVED** [review]

**Review by szaimen — APPROVED** [review]
no phpunit here?

**Comment by come-nc** [comment]
There are unit test but no phpunit in CI, I will add it.

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -0,0 +1,264 @@
+name: PHPUnit
```
Comment (nickvergessen) [inline .github/workflows/phpunit.yml:1]: Better to pick the one from app-tutorial:
https://github.com/nextcloud/app-tutorial/tree/master/.github/workflows

**Comment by nickvergessen** [comment]
> Run ./vendor/phpunit/phpunit/phpunit -c ./tests/phpunit.xml
PHP Warning:  require_once(PHPUnit/Autoload.php): failed to open stream: No such file or directory in /home/runner/work/privacy/privacy/apps/privacy/tests/bootstrap.php on line 16
PHP Fatal error:  require_once(): Failed opening required 'PHPUnit/Autoload.php' (include_path='/home/runner/work/privacy/privacy/3rdparty/pear/archive_tar:/home/runner/work/privacy/privacy/3rdparty/pear/console_getopt:/home/runner/work/privacy/privacy/3rdparty/pear/pear-core-minimal/src:/home/runner/work/privacy/privacy/3rdparty/pear/pear_exception:/home/runner/work/privacy/privacy/apps') in /home/runner/work/privacy/privacy/apps/privacy/tests/bootstrap.php on line 16
Error: Process completed with exit code 255.

**Comment by nickvergessen** [comment]
Added the summaries as required

**Comment by nickvergessen** [comment]
>  1) OCA\Privacy\Tests\Unit\Controller\PageControllerTest::testIndex
Error: Class 'OCA\Privacy\Controller\PageController' not found

The test files look pretty broken, maybe that is why it wasn't run at all :D

**Comment by come-nc** [comment]
> > 1. OCA\Privacy\Tests\Unit\Controller\PageControllerTest::testIndex
> >    Error: Class 'OCA\Privacy\Controller\PageController' not found
> 
> The test files look pretty broken, maybe that is why it wasn't run at all :D

PageControllerTest is the only non-empty test and there is no PageController class in the app :thinking:

**Comment by nickvergessen** [comment]
`rm -rf tests/*` ?

**Comment by come-nc** [comment]
> `rm -rf tests/*` ?

If you commit it I approve it :-P

File: .github/workflows/phpunit-mysql.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+      - name: Check file existence
+        id: check_files
+        uses: andstor/file-existence-action@v1
+        with:
+          files: ${{ env.PHPUNIT_CONFIG }}
+
+      - name: PHPUnit
+        # Only run if phpunit config file exists
+        if: steps.check_files.outputs.files_exists == 'true'
```
Comment (nickvergessen) [inline .github/workflows/phpunit-mysql.yml:88]: Seems to work, need the same for the integration one (but it exists in this repo) on the template and we are good to go I guess?

Comment (come-nc) [reply .github/workflows/phpunit-mysql.yml:87]: yep

File: .github/workflows/phpunit-mysql.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        working-directory: apps/${{ env.APP_NAME }}
+        run: ./vendor/phpunit/phpunit/phpunit -c ${{ env.PHPUNIT_CONFIG }}
+
+      - name: Check file existence
+        id: check_files
```
Comment (nickvergessen) [inline .github/workflows/phpunit-mysql.yml:92]: duplicate id?

**Comment by nickvergessen** [comment]
Works now, last sign of approval? @come-nc

**Comment by come-nc** [comment]
No integration tests are not run.

**Comment by come-nc** [comment]
```
The workflow is not valid. .github/workflows/phpunit-mysql.yml (Line: 82, Col: 9): Unexpected value 'working-directory' .github/workflows/phpunit-mysql.yml (Line: 95, Col: 9): Unexpected value 'working-directory'
￼Show more
```
:disappointed:

**Comment by come-nc** [comment]
Finally :partying_face: :fireworks:

**Pull request merged**